apps/src/lab2/views/dialogs/GenericConfirmationDialog.tsx

```diff
@@ -1,50 +1,36 @@
 import React from 'react';
 
-import Typography from '@cdo/apps/componentLibrary/typography';
-import commonI18n from '@cdo/locale';
+import GenericDialog, {GenericDialogProps} from './GenericDialog';
 
-import {BaseDialogProps} from './DialogManager';
-
-import moduleStyles from './confirm-dialog.module.scss';
+export type GenericConfirmationDialogProps = Required<
+  Pick<GenericDialogProps, 'title' | 'message'>
+> & {
+  handleConfirm: () => void;
+  handleCancel?: () => void;
+  confirmText?: string;
+};
 
 /**
  * Generic confirmation dialog used in Lab2 labs.
  * The title, message, and confirm button text can be customized.
  * If no confirm button text is provided, the default text is "OK" (translatable).
  */
-const GenericConfirmationDialog: React.FunctionComponent<BaseDialogProps> = ({
-  handleConfirm,
-  handleCancel,
-  title,
-  message,
-  confirmText,
-}) => {
-  return (
-    <div className={moduleStyles.confirmDialog}>
-      <Typography semanticTag="h1" visualAppearance="heading-lg">
-        {title}
-      </Typography>
-      <Typography semanticTag="p" visualAppearance="body-two">
-        {message}
-      </Typography>
-      <div className={moduleStyles.buttonContainer}>
-        <button
-          className={moduleStyles.cancel}
-          type="button"
-          onClick={handleCancel}
-        >
-          {commonI18n.cancel()}
-        </button>
-        <button
-          className={moduleStyles.confirm}
-          type="button"
-          onClick={handleConfirm}
-        >
-          {confirmText || commonI18n.dialogOK()}
-        </button>
-      </div>
-    </div>
-  );
-};
+const GenericConfirmationDialog: React.FunctionComponent<
+  GenericConfirmationDialogProps
+> = ({title, message, handleConfirm, handleCancel, confirmText}) => (
+  <GenericDialog
+    title={title}
+    message={message}
+    buttons={{
+      confirm: {
+        callback: handleConfirm,
+        text: confirmText,
+      },
+      cancel: {
+        callback: handleCancel,
+      },
+    }}
+  />
+);
 
 export default GenericConfirmationDialog;
```

Comment:
Likewise, the pre-existing components were just re-tooled to centralize around `GenericDialog`, and let that component handle all of the layout.